mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -648,4 +648,50 @@ mod tests {
 
         assert_eq!(expected_user_agent, user_agent_header_value);
     }
+
+    #[test]
+    #[ignore]
+    // Simple test to ensure the user agent header is correct even when given profile contains calid credentials.
+    fn test_client_with_valid_credentials() {
+        let expected_user_agent = "mountpoint-s3-client";
+        let profile_name_override = String::from("VALID_CREDENTIALS");
+
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+
+        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");
+
+        let mut message = client
+            .new_request_template("GET", "plutotestankit")
+            .expect("new request template expected");
+
+        let headers = message.inner.get_headers().expect("Expected a block of HTTP headers");
+
+        let user_agent_header = headers
+            .get("User-Agent")
+            .expect("User Agent Header expected with given prefix");
+        let user_agent_header_value = user_agent_header.value();
+
+        assert_eq!(expected_user_agent, user_agent_header_value);
+    }
+
+    #[test]
+    #[should_panic]
+    // Simple test with invalid credentials when creating S3CrtClient.
+    fn test_client_with_invalid_credentials() {
+        let profile_name_override = String::from("INVALID_CREDENTIALS");
+
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+
+        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");
+
+        let mut message = client
+            .new_request_template("GET", "plutotestankit")
+            .expect("new request template expected");
+    }
```

Comment:
When should this panic? During client creation?

We can use `expect_err` if we want to test if something is `Err` instead of `Ok`, rather than using `should_panic`.

Side note - the profile doesn't exist, right? Let's describe it that way in the string, I was wondering what credentials were invalid.

```suggestion
    #[test]
    // Simple test with invalid credentials when creating S3CrtClient.
    fn test_client_with_invalid_credentials() {
        let profile_name_override = String::from("NONEXISTENT_PROFILE");

        let config = S3ClientConfig {
            profile_name_override: Some(profile_name_override),
            ..Default::default()
        };

        let client = S3CrtClient::new("eu-west-1", config).expect("create test client");

        let mut message = client
            .new_request_template("GET", "some-bucket")
            .expect_err("request construction should fail");
    }
```